share/functions/fish_status_to_signal.fish

```diff
@@ -1,17 +1,10 @@
 function fish_status_to_signal --description "Print signal name from argument (\$status), or just argument"
+    __fish_make_completion_signals # Make sure signals are cached
     for arg in $argv
         if test $arg -gt 128
-            set -l signals SIGHUP SIGINT SIGQUIT SIGILL SIGTRAP SIGABRT SIGBUS \
-                SIGFPE SIGKILL SIGUSR1 SIGSEGV SIGUSR2 SIGPIPE SIGALRM \
-                SIGTERM SIGSTKFLT SIGCHLD SIGCONT SIGSTOP SIGTSTP \
-                SIGTTIN SIGTTOU SIGURG SIGXCPU SIGXFSZ SIGVTALRM \
-                SIGPROF SIGWINCH SIGIO SIGPWR SIGSYS
-            set -l sigix (math $arg - 128)
-            if test $sigix -le (count $signals)
-                echo $signals[$sigix]
-            else
-                echo $arg
-            end
+            # Important: each element of $__kill_signals is a string like "10 USR1"
+            string replace -r --filter "^$(math $arg-128) " SIG $__kill_signals |
+                read -l signal && echo $signal || echo $arg
```

Comment:
I think we can even say
`string replace -rf '^'(math $arg-128)' ' SIG $__kill_signals || echo $arg`

Reply:
You dare challenge my fish scripting prowess?!? 😄 At first glance I thought you were right, but then I tested it and remembered the problem (which I should have written a comment about). `$__kill_signals` is not deduplicated i.e `17 CHLD` and `17 CLD` are both elements. So we need some way of getting only one match, which `read` lets us do. 

We _can_ shorten it to
```fish
string replace -r --filter '^'(math $arg-128)' ' SIG $__kill_signals | read || echo $arg
```

Reply:
right, that's subtle :o

we don't need to do it in one line, how about

```fish
if set -l signal_names (string replace -r --filter '^'(math $arg-128)' ' SIG $__kill_signals)
    # Some signals have multiple mnemonics. Pick the first one.
    echo $signal_names[1]
else
    echo $arg
end
```

Reply:
> we don't need to do it in one line

Blasphemy, a true shell scripter does everything on one line.

Yah sure, we don't need to do it on one line, and I can change it if you'd like. If you use command substitution though, the whole function ends up being 20% slower, because command substitution is much slower than piping in general. And if you use `read`, well then you're basically where we are now, just more verbose.

I'd say just leave it the way it is, with an appropriate comment on the purpose of `read`. But your choice 👍

Reply:
I don't expect this perfomance difference to matter.
Also I'm surprised that command substitution is slower. I think they use the same machinery as when piping to builtins like `read`. We should just fix that. A benchmark might be helpful here.

Reply:
Very well, I'll make it multi-line.

Benchmark: https://github.com/fish-shell/fish-shell/issues/8542